### DISCUSSION TITLE
Unify releases version of all Tauri packages

### DISCUSSION THREAD

**Discussion by martpie** [tauri-apps/tauri#8931] [Ideas]
This may have been discussed before, but I did not find any similar discussion. **This is a convenience issue**, not a hard-blocker.

When Tauri packages are released (both Crates and npm packages), the different versions for different packages within the same release make it hard to understand which version should go with which version.

Typical release:

![Screenshot 2024-02-21 at 13 34 26](https://github.com/tauri-apps/tauri/assets/1311607/851ddbf7-04b8-4ba0-9957-fbb4c84e7c2f)

There's a `beta.1`, `beta.2`, `beta.3` in the same release, from here, I can understand they're from the same "release", but from a  `cargo.toml` it is more than tedious.

![Screenshot 2024-02-21 at 13 51 44](https://github.com/tauri-apps/tauri/assets/1311607/f50b5758-5ec0-47ab-a4c9-0ae3c7e307db)

![Screenshot 2024-02-21 at 13 53 04](https://github.com/tauri-apps/tauri/assets/1311607/b871b6f2-aa14-4786-bde4-6faafaa23d05)

**_Has the Tauri team thought about using the same version number for each release?_**

Tauri updates are rather rare (every other month maybe), and bring many improvements, so having unified version across the official ecosystem would make things like "how late am i in my dependencies, and what changed since?" quite easier.

I understand that switching to this new system would make that some package updates are "empty" (that's the downside of this approach).

I am just curious about the Tauri's and community's thought on this :)

**Comment by FabianLars** [comment]
In v1 we've always matched the minor release for all packages (minus 1.6 a few days ago...) and for v2 pre release versions we (or at least i did) always expect users to use the latest versions of all packages and commit their lockfiles to a git repo since everything can and will break on every other release, sometimes requiring multiple releases of a few crates in quick succession to fix something.
Since cargo and npm can upgrade the versions with a simple command i considered this simple enough for the user == not worth the hassle for us tbh.

Anywayyy, we're so close to rc/stable that it doesn't matter at this point and i'm just responding so you don't think we ignore you/the discussion section 😅 
afaik we won't unify versions once stable hits either because being able to individually update the features was one of the reasons why we moved most of the functionality into seperate plugins.

One thing i'd love to see in the future is some check in the cli that makes sure the npm and rust dependencies of a plugin (or core itself) are compatible though.

**Reply by martpie** [reply]
Absolutely fair :)
